### PR TITLE
Task3.1 main app mongodb cucumber

### PR DESCRIPTION
**Pull request by GuramSalia**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -100,10 +100,10 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-aop</artifactId>
         </dependency>
-<!--        <dependency>-->
-<!--            <groupId>io.github.resilience4j</groupId>-->
-<!--            <artifactId>resilience4j-spring-boot3</artifactId>-->
-<!--        </dependency>-->
+        <!--        <dependency>-->
+        <!--            <groupId>io.github.resilience4j</groupId>-->
```
Comment (SiarheiBor) [inline pom.xml]: delete unused comment

File: src/main/java/com/epam/gym_app_main_mq/model/Trainee.java [hunk 2]
```diff
@@ -69,8 +69,4 @@ public boolean equals(Object o) {
     @Override
     public int hashCode() {return Objects.hash(getUserId(), getDob(), getAddress());}
 
-    //    @Override
```
Comment (SiarheiBor) [inline src/main/java/com/epam/gym_app_main_mq/model/Trainee.java:72]: remove comment

File: src/main/java/com/epam/gym_app_main_mq/model/Trainer.java [hunk 3]
```diff
@@ -46,8 +46,4 @@ public boolean equals(Object o) {
     @Override
     public int hashCode() {return Objects.hash(getUserId());}
 
-    //    @Override
```
Comment (SiarheiBor) [inline src/main/java/com/epam/gym_app_main_mq/model/Trainer.java:49]: remove comment

File: src/main/java/com/epam/gym_app_main_mq/service/TrainingService.java [hunk 4]
```diff
@@ -33,12 +33,12 @@ public TrainingService(
...
     public Training getById(int id) {
-        log.info(">>>> Getting training with id: " + id);
+        log.info(">>>> Getting training with id: {}", id);
 
         Optional<Training> trainingOptional = trainingDAO.getById(id);
         if (trainingOptional.isEmpty()) {
-            log.error("Cannot get training with id: " + id);
-            throw new TrainingNotFoundException("training not found");
+            log.error("Cannot get training with id: {}", id);
+//            throw new TrainingNotFoundException("training not found");
```
Comment (SiarheiBor) [inline src/main/java/com/epam/gym_app_main_mq/service/TrainingService.java]: remove comment

File: src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+    private String correlationIdGetFullStat;
+
+    @Autowired
+    public IntegrationSteps(
+            Senders senders,
+            JmsTemplate jmsTemplate,
+            TestHelperRepository testHelperRepository
+    ) {
+        this.senders = senders;
+//        this.jmsTemplate = jmsTemplate;
```
Comment (SiarheiBor) [inline src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java]: remove comment

File: src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+        boolean messageReceived = false;
+        long timeout = System.currentTimeMillis() + 3000;
+
+        while (System.currentTimeMillis() < timeout) {
+
+            TestCorrelationIdHolder testCorrelationIdHolder = testHelperRepository.findById(1).orElse(null);
+            assert testCorrelationIdHolder != null;
+            String receivedCorrelationId = testCorrelationIdHolder.getStatUpdateCorrelationId();
+
+//            if (receivedCorrelationId.equals(this.correlationIdUpdateStat)) {
```
Comment (SiarheiBor) [inline src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java]: remove comment

File: src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+            }
+        }
+
+        assert messageReceived : "App 1 did not receive the expected response";
+    }
+
+    @Given("App 1 sends a full stat request with correlation ID {string}")
+    public void appSendsAFullStatRequestWithCorrelationID(String correlationId) {
+//        TestCorrelationIdHolder testCorrelationIdHolder = testHelperRepository.findById(1).orElse(null);
+//        assert testCorrelationIdHolder != null;
```
Comment (SiarheiBor) [inline src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java]: remove comment

File: src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+        long timeout = System.currentTimeMillis() + 3000;
+
+        while (System.currentTimeMillis() < timeout) {
+
+            TestCorrelationIdHolder testCorrelationIdHolder = testHelperRepository.findById(1).orElse(null);
+            assert testCorrelationIdHolder != null;
+
+            String receivedCorrelationId = testCorrelationIdHolder.getFullStatCorrelationId();
+//            if (receivedCorrelationId.equals(this.correlationIdGetFullStat)) {
+//                messageReceived = true;
```
Comment (SiarheiBor) [inline src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java]: remove comment

File: src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+        }
+
+        assert messageReceived : "App 1 did not receive the expected response";
+    }
+
+    @Given("App 1 sends a monthly request with correlation ID {string}")
+    public void appSendsAMonthlyRequestWithCorrelationID(String correlationId) {
+//        TestCorrelationIdHolder testCorrelationIdHolder = testHelperRepository.findById(1).orElse(null);
+//        assert testCorrelationIdHolder != null;
+//        log.info("testCorrelationId monthly-stat: {}", testCorrelationIdHolder.getMonthlyStatCorrelationId());
```
Comment (SiarheiBor) [inline src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java]: remove comment

File: pom.xml (hunk 1 above)
Comment (GuramSalia) [reply pom.xml]: removed

File: src/main/java/com/epam/gym_app_main_mq/service/TrainingService.java (hunk 4 above)
Comment (GuramSalia) [reply src/main/java/com/epam/gym_app_main_mq/service/TrainingService.java]: removed

File: src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java (hunk 5 above)
Comment (GuramSalia) [reply src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java]: removed

File: src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java (hunk 6 above)
Comment (GuramSalia) [reply src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java]: removed

File: src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java (hunk 7 above)
Comment (GuramSalia) [reply src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java]: removed

File: src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java (hunk 8 above)
Comment (GuramSalia) [reply src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java]: removed

File: src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java (hunk 9 above)
Comment (GuramSalia) [reply src/test/java/com/epam/gym_app_main_mq/integration/IntegrationSteps.java]: removed

**Pull request merged**